OSDOCS#8034: additional custom security groups at ROSA cluster creation time (#66328)
openshift/openshift-docs: modules/rosa-required-aws-service-quotas.adoc

[id="rosa-required-aws-service-quotas_{context}"]
= Required AWS service quotas

> The table below describes the AWS service quotas and levels required to create and run one {product-title} cluster. While most default values are suitable for most workloads, you may need to request additional quota for the following cases:

Reviewer: Per ISG
```suggestion
The table below describes the AWS service quotas and levels required to create and run one {product-title} cluster. Although most default values are suitable for most workloads, you might need to request additional quota for the following cases:
```

> The table below describes the AWS service quotas and levels required to create and run one {product-title} cluster. While most default values are suitable for most workloads, you are likely to need to request additional quota for the following cases:

Reviewer 2: Minor suggestion

```suggestion
The table below describes the AWS service quotas and levels required to create and run one {product-title} cluster. While most default values are suitable for most workloads, you might need to request additional quota for the following cases:
```

Author: Edited, thankyou

> * ROSA clusters require at least 100 vCPUs, but the default maximum value for vCPUs assigned to Running On-Demand Standard Amazon EC2 instances is `5`. Therefore if you have not created a ROSA cluster using the same AWS account previously, you must request additional EC2 quota for `Running On-Demand Standard (A, C, D, H, I, M, R, T, Z) instances`.

Reviewer 2: It's unclear to me why the last part here is in backticks. Is this some literal value somewhere (CLI/UI)? I don't see any references of anything like that anywhere.

Author: @bergerhoffer It's a label / field and a specific quota type in the AWS Console, which I think is best kept as a literal value.

> * Some optional cluster configuration features, such as custom security groups, may require that you request additional quota. For example, because ROSA uses 3 security groups by default, and the default quota for `Security groups per network interface` is `5`, up to 2 custom security groups can be added to a cluster without requesting additional quota for `Security groups per network interface`.

Reviewer 3: This isn't quite right.

We create 3 security groups and attach them to the VPC but not the network interfaces. A network interface is created and associated with each new worker machine pool, but we only attach the worker-security-group to each worker machine pool and the user would have 4 extra security groups to use without any changes to AWS quota.

Author: Thanks @gdbranco! Based on the new maximum of 5 additional security groups this would mean that users can add up to 4 additional SGs without changing quota, and only need to increase quota if they want to add a 5th additional SG, correct?

Author: I've reworded this to the following:

* Some optional cluster configuration features, such as custom security groups, may require that you request additional quota. For example, because ROSA associates the `worker-security-group` security group with network interfaces in worker machine pools by default, and the default quota for `Security groups per network interface` is `5`, up to 4 additional custom security groups for worker nodes can be added to the cluster without requesting additional quota for `Security groups per network interface`.

Author: CC @bchandra-ocp @gdbranco Please confirm whether we are supporting up to 5 security groups total or up to 5 additional security groups

Author: Confirmed, 5 additional.

> * Some optional cluster configuration features, such as custom security groups, may require you to request additional quota. For example, because ROSA associates 1 security group with network interfaces in worker machine pools by default, and the default quota for `Security groups per network interface` is `5`, you can add up to 4 additional custom security groups for worker nodes without requesting additional quota for `Security groups per network interface`.

Reviewer: ```suggestion
* Some optional cluster configuration features, such as custom security groups, might require that you request additional quota. For example, because ROSA associates 1 security group with network interfaces in worker machine pools by default, and the default quota for `Security groups per network interface` is `5`, you can add up to 4 additional custom security groups for worker nodes without requesting additional quota for `Security groups per network interface`.
```

> * Some optional cluster configuration features, such as custom security groups, may require that you request additional quota. For example, because ROSA associates 1 security group with network interfaces in worker machine pools by default, and the default quota for `Security groups per network interface` is `5`, up to 4 additional custom security groups for worker nodes can be added to the cluster without requesting additional quota for `Security groups per network interface`.

Reviewer 2: Some minor suggestions

```suggestion
* Some optional cluster configuration features, such as custom security groups, might require you to request additional quota. For example, because ROSA associates 1 security group with network interfaces in worker machine pools by default, and the default quota for `Security groups per network interface` is `5`, you can add up to 4 additional custom security groups for worker nodes to the cluster without requesting additional quota for `Security groups per network interface`.
```

Author: Done, thankyou!

If you need to modify or increase a specific quota, see Amazon's documentation on link:https://docs.aws.amazon.com/servicequotas/latest/userguide/request-quota-increase.html[requesting a quota increase]. Large quota requests are submitted to Amazon Support for review, and take some time to be approved. If your quota request is urgent, contact AWS Support.

.ROSA-required service quota

[options="header"]
|===

> |Quota name |Service code |Quota code| AWS Default | Minimum required | Description

Reviewer 2: ```suggestion
|Quota name |Service code |Quota code| AWS default | Minimum required | Description
```

[options="header"]
|===

> |Quota name |Service code |Quota code| AWS Default | Minimum required | Description

Reviewer 2: ```suggestion
|Quota name |Service code |Quota code| AWS default | Minimum required | Description
```

Author: Done, thankyou!

|L-53DA6B97
|50
|50

> |The maximum number of Application Load Balancers per Region

Reviewer 2: Just checking that this needs to still be in title case in the description?

Author: Application Load Balancers does, to be distinct from Classic Load Balancers, but the rest doesn't - thanks!

|Classic Load Balancers per Region
|elasticloadbalancing
|L-E9E9831D
|20
|20

> |The maximum number of Classic Load Balancers per Region

Reviewer 2: Same question on title case or not

Author: as above
